packages/blade/src/components/Amount/Amount.tsx

```diff
@@ -0,0 +1,271 @@
+   */
+  intent?: Exclude<Feedback, 'neutral'> | undefined;
+  /**
+   * Sets the size of the amount.
+   *
+   * @default 'heading'
+   */
+  size?:
+    | `title-${NonNullable<Exclude<TitleProps['size'], 'large'>>}`
+    | `heading-${NonNullable<Exclude<HeadingProps<{ variant: 'regular' }>['size'], 'medium'>>}`
+    | `heading-${NonNullable<
+        Exclude<HeadingProps<{ variant: 'regular' }>['size'], 'medium'>
+      >}-${NonNullable<Exclude<TextProps<{ variant: 'body' }>['weight'], 'regular'>>}`
+    | `body-${NonNullable<Exclude<TextProps<{ variant: 'body' }>['size'], 'xsmall'>>}`
+    | `body-${NonNullable<Exclude<TextProps<{ variant: 'body' }>['size'], 'xsmall'>>}-${NonNullable<
+        Exclude<TextProps<{ variant: 'body' }>['weight'], 'regular'>
+      >}`;
+  /**
+   * Indicates what the suffix of amount should be
+   *
+   * @default 'decimals'
+   */
+  suffix?: 'decimals' | 'none' | 'humanize';
+  /**
+   * Highlights the main amount by making the prefix symbol and decimal digits subtle
+   *
+   * @default true
+   */
+  isAffixSubtle?: true | false;
+  /**
+   * Prefix to be shown before the amount value. The prefix can be either a currency symbol or a currency code.
+   *
+   * @default 'currency-symbol'
+   */
+  prefix?: 'currency-symbol' | 'currency-code';
+} & TestID;
+
+type ColorProps = {
+  amountValueColor: BaseTextProps['color'];
+  affixColor: BaseTextProps['color'];
+};
+
+const getTextColorProps = ({ intent }: { intent: AmountProps['intent'] }): ColorProps => {
+  const props: ColorProps = {
+    amountValueColor: 'surface.text.normal.lowContrast',
+    affixColor: 'surface.text.muted.lowContrast',
+  };
+  if (!intent) return props;
+  props.amountValueColor = `feedback.text.${intent}.lowContrast`;
+  props.affixColor = `feedback.text.${intent}.lowContrast`;
+  return props;
+};
+
+const getAffixFontWeight = (isAffixSubtle: true | false): 'regular' | 'bold' => {
+  if (isAffixSubtle) return 'regular';
+  return 'bold';
+};
+
+const getMainValueWeight = (
+  size: NonNullable<AmountProps['size']>,
+): NonNullable<TextProps<{ variant: 'body' }>['weight']> => {
+  return size.includes('bold') || size.startsWith('title') ? 'bold' : 'regular';
+};
+
+const getAffixFontSize = (
+  isAffixSubtle: NonNullable<AmountProps['isAffixSubtle']>,
+  size: NonNullable<AmountProps['size']>,
+): keyof FontSize | undefined => {
+  if (isAffixSubtle) return affixFontSizes[size];
+  return amountTextSizes[size];
+};
+
+interface AmountValue extends Omit<AmountProps, 'value'> {
+  affixColor: BaseTextProps['color'];
+  amountValueColor: BaseTextProps['color'];
+  value: string;
+}
+
+const AmountValue = ({
+  value,
+  size = 'body-medium',
+  amountValueColor,
+  isAffixSubtle = true,
+  suffix = 'decimals',
+  affixColor,
+}: AmountValue): ReactElement => {
+  const affixFontWeight = getAffixFontWeight(isAffixSubtle);
+  const affixFontSize = getAffixFontSize(isAffixSubtle, size);
+  const valueForWeight = getMainValueWeight(size);
```

Comment:
These can be direct inline checks since there's nothing complex happening here

```suggestion
  const affixFontWeight = isAffixSubtle ? 'regular' : 'bold';
  const affixFontSize = isAffixSubtle ? affixFontSizes[size] : amountTextSizes[size];;
  const valueForWeight = size.includes('bold') || size.startsWith('title') ? 'bold' : 'regular';
```